units/latest.pxu

```diff
@@ -1,104 +1,6 @@
-category_id: edgex
-flags: simple
-
-id: edgex/latest/refresh-config-paths
-_summary: Test that refreshing uses "current" based filepaths in config files
-command: $PLAINBOX_PROVIDER_DATA/latest/test-refresh-config-paths.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/refresh-services
-_summary: Test that services start after refreshing 
-command: $PLAINBOX_PROVIDER_DATA/latest/test-refresh-services.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/self-refresh-services
-_summary: Test that services start after refreshing to this revision from self
-command: $PLAINBOX_PROVIDER_DATA/latest/test-self-refresh-services.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/sys-mgmt-agent
+id: edgex/latest/run-all-tests-locally
 _summary: Test that the system management agent works with the snap
-command: $PLAINBOX_PROVIDER_DATA/latest/test-sys-mgmt-agent.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/network-interfaces
-_summary: Test that services are not listening on external network interfaces
-command: $PLAINBOX_PROVIDER_DATA/latest/test-network-interfaces.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/rules-engine
-_summary: Test that the rules engine works with the snap
-command: $PLAINBOX_PROVIDER_DATA/latest/test-rules-engine.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/refresh-core-18-20
-_summary: Test that Kong configuration is migrated when upgrading from core 18 to 20
-command: $PLAINBOX_PROVIDER_DATA/latest/test-refresh-core-18-20.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/config-hook
-_summary: Test that snap configure hook settings are supported by the edgexfoundry snap
-command: $PLAINBOX_PROVIDER_DATA/latest/test-config-hook.sh
-user: root
-environ: DEFAULT_TEST_CHANNEL REVISION_TO_TEST REVISION_TO_TEST_CHANNEL REVISION_TO_TEST_CONFINEMENT
-category_id: edgex
-flags: simple
-
-id: edgex/latest/device-virtual
-_summary: Test that device-virtual creates devices and produces readings
-command: $PLAINBOX_PROVIDER_DATA/latest/test-device-virtual.sh
+command: $PLAINBOX_PROVIDER_DATA/latest/run-all-tests-locally.sh
```

Comment:
We need to run three tests:
- edgexfoundry
- edgexfoundry/refresh (for EdgeX >3 upgrades)
- edgex-no-sec

Squashing all into one bash script and plan is not a good idea because the checkbox CLI would lose control of the individual plans and the test results will be hard to read.

Reply:
I suggest having one script which takes the name of the go suite as argument and runs it. For example `run-go-tests.sh edgex-no-sec`:
1. If snap-testing directory exists, delete it
2. Shallow clone v3 snap-testing
3. export PLATFORM_CHANNEL=$DEFAULT_TEST_CHANNEL
4. Run the test suite 
5. If tests failed / non-zero exit code, then print the service error logs (.log file) to standard output. See https://github.com/canonical/edgex-snap-testing/blob/042c164fa2c074a0b4fd8ee6de3b906fac23a0e2/test/action.yml#L72

We don't wanna delete in the end because that removes the logs.